File: Ejercicios Propuestos 3/Ejercicio 12/Ejercicio 12.py
NAME = "Renzo Leonardo Cienfuegos Cardenas"
GROUP = 2

# Variables
suma = 0
cantidad = 0

# Abrir el archivo en modo lectura
with open("estudiantes.txt", "r", encoding="utf-8") as archivo:

    # Leer cada línea del archivo
    for linea in archivo:

        # Separar el nombre y la nota
        datos = linea.split()

        nombre = datos[0]
        nota = float(datos[1])

        print(f"Estudiante: {nombre}  Nota: {nota}")

        suma += nota
        cantidad += 1

# Calcular el promedio
if cantidad > 0:
    promedio = suma / cantidad
    print("\nPromedio general:", promedio)
else:
    print("El archivo está vacío.")

File: Ejercicios Propuestos 3/Ejercicio 12/Ejercicio 12.jl
nombre = ""
nota = 0.0
suma = 0.0
promedio = 0.0
cantidad = 0

# Abrir el archivo para lectura
archivo = open("Estudiantes.txt", "r")

# Leer el archivo línea por línea
for linea in eachline(archivo)

    datos = split(linea)

    nombre = datos[1]
    nota = parse(Float64, datos[2])

    println("Estudiante: ", nombre, "  Nota: ", nota)

    global suma += nota
    global cantidad += 1

end

# Cerrar el archivo
close(archivo)

# Calcular el promedio
if cantidad > 0
    promedio = suma / cantidad
    println()
    println("Promedio general: ", promedio)
else
    println("El archivo está vacío.")
end
